### PR TITLE
PR: Add IconWidget to display icon as widget

### PR DESCRIPTION
**Pull request by doudz**
Add IconWidget object to display any icon as a widget

```python
# Spining icon as a widget
spin_widget = qta.IconWidget()
spin_icon = qta.icon('mdi.loading', color='red',
                     animation=qta.Spin(spin_widget))
spin_widget.setIcon(spin_icon)

# simple widget
simple_widget = qta.IconWidget('mdi.web', color='blue')
```

qtawesome-screenshot

### CODE REVIEW

**Comment by ccordoba12** [comment]
@goanpeca, what do you think as this one?

**Comment by doudz** [comment]
I improved it a little

**Comment by goanpeca** [comment]
Sure, it can be useful on some cases.

Thanks for working on this 👍🏼

File: qtawesome/__init__.py [hunk 1]
```diff
@@ -222,3 +223,26 @@ def set_defaults(**kwargs):
...
     return set_global_defaults(**kwargs)
 
+
+class IconWidget(QtWidgets.QLabel):
+    '''
+    IconWidget gives the ability to display an icon as a widget
+    it supports setIcon() and setIconSize()
+    '''
+    def __init__(self, *names, **kwargs):
+        super().__init__()
```
Comment (ccordoba12) [inline qtawesome/__init__.py]: This is not the correct usage of `super` in Python 2.

Comment (goanpeca) [reply qtawesome/__init__.py]:
```suggestion
        super(IconWidget, self).__init__()
```

File: qtawesome/__init__.py [hunk 2]
```diff
@@ -222,3 +223,26 @@ def set_defaults(**kwargs):
...
+        self._icon = None
+        self._size = QtCore.QSize(16, 16)
+        self.setIcon(icon(*names, **kwargs))
+
+    def setIcon(self, _icon):
```
Comment (goanpeca) [inline qtawesome/__init__.py]: Please add a docstring

File: qtawesome/__init__.py [hunk 3]
```diff
@@ -222,3 +223,26 @@ def set_defaults(**kwargs):
...
+        self._icon = _icon
+        self.setPixmap(_icon.pixmap(self._size))
+
+    def setIconSize(self, size):
+        self._size = size
```
Comment (goanpeca) [inline qtawesome/__init__.py]: Please add a docstring

File: qtawesome/__init__.py [hunk 4]
```diff
@@ -222,3 +223,26 @@ def set_defaults(**kwargs):
...
+
+    def update(self, *args, **kwargs):
```
Comment (goanpeca) [inline qtawesome/__init__.py]: Please add a docstring

File: qtawesome/__init__.py [hunk 5]
```diff
@@ -222,3 +223,26 @@ def set_defaults(**kwargs):
...
+        if self._icon:
+            self.setPixmap(self._icon.pixmap(self._size))
+        return super().update(*args, **kwargs)
```
Comment (goanpeca) [inline qtawesome/__init__.py]:
```suggestion
        return super(IconWidget, self).update(*args, **kwargs)
```

File: README.md [hunk 6]
```diff
@@ -123,6 +123,19 @@ spin_icon = qta.icon('fa5s.spinner', color='red',
...
+- Display Icon as a widget
+
+```python
+# Spining icon as a widget
+spin_widget = qta.IconWidget()
+spin_icon = qta.icon('mdi.loading', color='red',
+                     animation=qta.Spin(spin_widget))
+spin_widget.setIcon(spin_icon)
+
+# simple widget
```
Comment (goanpeca) [inline README.md]:
```suggestion
# Simple icon widget
```

File: README.md [hunk 7]
```diff
@@ -123,6 +123,19 @@ spin_icon = qta.icon('fa5s.spinner', color='red',
 spin_button.setIcon(spin_icon)
 ```
 
+- Display Icon as a widget
+
+```python
+# Spining icon as a widget
```
Comment (goanpeca) [inline README.md]:
```suggestion
# Spining icon widget spinning example
```

**Comment by doudz** [comment]
Sorry for delay
I think it's ready now

**Comment by ccordoba12** [comment]
@goanpeca, please give a new review to this one and approve if it's ready.

**Comment by goanpeca** [comment]
Please merge with master so new CI can run. Thanks! I will review after that :-)

**Comment by goanpeca** [comment]
@doudz could you merge with master and solve the conflicts?

**Comment by ccordoba12** [comment]
@doudz, please merge with master to fix the new merge conflict.

**Comment by ccordoba12** [comment]
@goanpeca, is this ready?

File: qtawesome/__init__.py [hunk 8]
```diff
@@ -222,3 +223,49 @@ def set_defaults(**kwargs):
     """
     return set_global_defaults(**kwargs)
 
+
+class IconWidget(QtWidgets.QLabel):
+    '''
```
Comment (ccordoba12) [inline qtawesome/__init__.py]: @doudz, please use double quotes (`"`) instead of single ones (`'`) for all docstrings in this class.

**Pull request merged**